server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+      switch (type) {
+        case ADDRESS:
+        case EMAIL:
+        case ID:
+        case NAME:
+        case TEXT:
+          return contextualizedScalarsJsonPresenter;
+
+          // Answers to enumerator questions are not be included because the path is
+          // incompatible with the JSON export schema. This is because enumerators store an
+          // identifier
+          // value for each repeated entity, which with the current export logic conflicts with the
+          // answers stored for repeated entities.
+        case ENUMERATOR:
+
+          // Static content questions are not included in API responses because they
+          // do not include an answer from the user.
+        case STATIC:
+          return emptyJsonPresenter;
+
+        case CHECKBOX:
+          return multiSelectJsonPresenter;
+        case FILEUPLOAD:
+          return fileUploadJsonPresenter;
+        case NUMBER:
+          return numberJsonPresenter;
+        case PHONE:
+          return phoneJsonPresenter;
+        case RADIO_BUTTON:
+        case DROPDOWN:
+          return singleSelectJsonPresenter;
+        case CURRENCY:
+          return currencyJsonPresenter;
+        case DATE:
+          return dateJsonPresenter;
+
+        default:
+          throw new RuntimeException(String.format("Unrecognized questionType %s", type));
+      }
+    }
+  }
+
+  class FileUploadJsonPresenter implements QuestionJsonPresenter<FileUploadQuestion> {
+
+    private final String baseUrl;
+
+    @Inject
+    FileUploadJsonPresenter(Config config) {
+      baseUrl = config.getString("base_url");
+    }
+
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(FileUploadQuestion question) {
+      return ImmutableMap.of(
+          question.getApplicantQuestion().getContextualizedPath().join(Scalar.FILE_KEY),
+          question
+              .getApplicantQuestion()
+              .createFileUploadQuestion()
+              .getFileKeyValue()
+              .map(
+                  fileKey ->
+                      baseUrl
+                          + controllers.routes.FileController.acledAdminShow(
+                                  URLEncoder.encode(fileKey, StandardCharsets.UTF_8))
+                              .url())
+              .orElse(""));
+    }
+  }
+
+  class MultiSelectJsonPresenter implements QuestionJsonPresenter<MultiSelectQuestion> {
+
+    @Override
+    public ImmutableMap<Path, ImmutableList<String>> getJsonEntries(MultiSelectQuestion question) {
+      Path path = question.getSelectionPath().asApplicationPath();
+
+      if (question.getSelectedOptionsValue().isPresent()) {
+        ImmutableList<String> selectedOptions =
+            question.getSelectedOptionsValue().get().stream()
+                .map(LocalizedQuestionOption::optionText)
+                .collect(ImmutableList.toImmutableList());
+
+        return ImmutableMap.of(path, selectedOptions);
+      }
+
+      return ImmutableMap.of();
+    }
+  }
```

Comment:
Above code is taken from `case CHECKBOX`:

https://github.com/civiform/civiform/blob/358ad3dbb6e67aacb91afb3ede60e32ac281ba26/server/app/services/export/JsonExporter.java#L121-L219